### PR TITLE
(doc) Improve wording in "Libraries with Dune" tutorial

### LINKED ISSUE #2034: Fix sentence on relationship between source files and modules
Tutorial: [Libraries with Dune](https://ocaml.org/docs/libraries-dune#minimum-project-setup)
File: [1ms_02_dune.md](../tree/main/blob/data/tutorials/language/1ms_02_dune.md), line 86

> In OCaml, each source file is compiled into a module.

* “Source file” is wrong, it should be ".ml file"
* “compiled into a module” is flummox. Let's reuse words from section [Modules and the file system](https://v2.ocaml.org/releases/5.1/htmlman/comp.html#s%3Amodules-file-system) of the manual

### PR DESCRIPTION
**Pull request by The-Amoghavarsha**
Closes #2034

### CODE REVIEW

File: data/tutorials/language/0it_05_imperative.md [hunk 1]
```diff
@@ -526,6 +526,18 @@ The following example uses the `get_char` function we defined earlier (in the se
...
 
+### Closures
+
+A closure is a function bundled together with its surrounding state, including any variables or references from the enclosing scope that it references. This bundled function-state package allows for the creation of self-contained units of behavior that can maintain internal state across multiple invocations. 
+
+```ocaml
+let counter () =
+  let n = ref 0 in
+  fun () -> incr n; !n;;
+```
```
Comment (christinerose) [inline data/tutorials/language/0it_05_imperative.md]: We've been formatting toplevel code examples with a `#` at the beginning and the expected return beneath the code. Shall I change it to match?

File: data/tutorials/language/0it_05_imperative.md [hunk 2]
```diff
@@ -526,6 +526,18 @@ The following example uses the `get_char` function we defined earlier (in the se
...
+
+In the above code snippet, a function `counter ()` is defined. The function creates a closure that keeps track of a counter `n` using a reference. Each time the closure is called, it increments the counter n by one and returns its current value.
```
Comment (christinerose) [inline data/tutorials/language/0it_05_imperative.md]:
```suggestion
In the above code snippet, a function `counter ()` is defined. The function creates a closure that keeps track of a counter `n` using a reference. Each time the closure is called, it increments the counter `n` by one and returns its current value. 
```

File: data/tutorials/language/1ms_02_dune.md [hunk 3]
```diff
@@ -83,7 +83,7 @@ Dune stores the files it creates in a folder named `_build`. In a project manage
 $ echo _build >> .gitignore
 ```
 
-In OCaml, each source file is compiled into a module. In the `mixtli` project, the file `cloud.ml` creates a module named `Cloud`, the file `wmo.ml` creates a module `Wmo` that contains two submodules: `Stratus` and `Cumulus`.
+In OCaml, each .ml file corresponds to a module. In the `mixtli` project, the file `cloud.ml` creates a module named `Cloud`, the file `wmo.ml` creates a module `Wmo` that contains two submodules: `Stratus` and `Cumulus`.
```
Comment (christinerose) [inline data/tutorials/language/1ms_02_dune.md]:
```suggestion
In OCaml, each `.ml` file corresponds to a module. In the `mixtli` project, the file `cloud.ml` creates a module named `Cloud`, and the file `wmo.ml` creates a module `Wmo` that contains two submodules: `Stratus` and `Cumulus`.
```

**Review by christinerose — COMMENTED** [review]
Made two formatting changes. One remaining question about code formatting.

File: data/tutorials/language/0it_05_imperative.md (hunk 1 above)
Comment (The-Amoghavarsha) [reply data/tutorials/language/0it_05_imperative.md]: Oh, I see. Thanks for clarifying. :smiley:

File: data/tutorials/language/0it_05_imperative.md
Comment (cuihtlauac) [inline data/tutorials/language/0it_05_imperative.md:1]: This is part of PR #2046. It shouldn't be part of this PR, @The-Amoghavarsha, can you remove it please?

**Review by cuihtlauac — CHANGES_REQUESTED** [review]

**Comment by cuihtlauac** [comment]
Also, checks are failing because the formatting was broken upstream. Please rebase.

File: data/tutorials/language/1ms_02_dune.md (hunk 3 above)
Comment (cuihtlauac) [inline data/tutorials/language/1ms_02_dune.md]:
```suggestion
In OCaml, each `.ml` file defines a module. In the `mixtli` project, the file `cloud.ml` defines the `Cloud` module, the file `wmo.ml` defines the `Wmo` module that contains two submodules: `Stratus` and `Cumulus`.
```

**Review by cuihtlauac — CHANGES_REQUESTED** [review]

**Review by cuihtlauac — APPROVED** [review]
